src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -968,62 +968,67 @@ void IslandWindow::_SetIsFullscreen(const bool fullscreenEnabled)
 }
 
 // Method Description:
-// - Call UnregisterHotKey once for each entry in hotkeyList, to unset all the bound global hotkeys.
-// Arguments:
-// - hotkeyList: a list of hotkeys to unbind
+// - Call UnregisterHotKey once for each previously registered hotkey.
 // Return Value:
 // - <none>
-void IslandWindow::UnsetHotkeys(const std::vector<winrt::Microsoft::Terminal::Control::KeyChord>& hotkeyList)
+void IslandWindow::UnregisterAllHotKeys()
 {
-    TraceLoggingWrite(g_hWindowsTerminalProvider,
-                      "UnsetHotkeys",
-                      TraceLoggingDescription("Emitted when clearing previously set hotkeys"),
-                      TraceLoggingInt64(hotkeyList.size(), "numHotkeys", "The number of hotkeys to unset"),
-                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
-                      TraceLoggingKeyword(TIL_KEYWORD_TRACE));
-
-    for (int i = 0; i < ::base::saturated_cast<int>(hotkeyList.size()); i++)
+    TraceLoggingWrite(
+        g_hWindowsTerminalProvider,
+        "UnregisterAllHotKeys",
+        TraceLoggingDescription("Emitted when clearing previously set hotkeys"),
+        TraceLoggingInt32(_nextHotkeyIndex, "numHotkeys", "The number of hotkeys to unset"),
+        TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+        TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+
+    for (; _nextHotkeyIndex != 0; --_nextHotkeyIndex)
     {
-        LOG_IF_WIN32_BOOL_FALSE(UnregisterHotKey(_window.get(), i));
+        LOG_IF_WIN32_BOOL_FALSE(::UnregisterHotKey(_window.get(), _nextHotkeyIndex));
     }
 }
 
 // Method Description:
-// - Call RegisterHotKey once for each entry in hotkeyList, to attempt to
-//   register that keybinding as a global hotkey.
+// - Call RegisterHotKey to attempt to register that keybinding as a global hotkey.
 // - When these keys are pressed, we'll get a WM_HOTKEY message with the payload
 //   containing the index we registered here.
+// - Call UnregisterAllHotKeys() before registering your hotkeys.
+//   See: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerhotkey#remarks
 // Arguments:
-// - hotkeyList: a list of hotkeys to bind
+// - hotkey: The key-combination to register.
 // Return Value:
 // - <none>
-void IslandWindow::SetGlobalHotkeys(const std::vector<winrt::Microsoft::Terminal::Control::KeyChord>& hotkeyList)
+void IslandWindow::RegisterHotKey(const winrt::Microsoft::Terminal::Control::KeyChord& hotkey)

```

Comment:
The idea behind the change to the interface of `IslandWindow::RegisterHotKey` is twofold:
* By passing the hotkey not as a vector but rather one by one, we can get rid of the `std::vector` of key chords in the AppHost
* This allows IslandWindow to manage the hotkey ID itself! With `_nextHotkeyIndex` in place it's impossible to accidentally call `RegisterHotKey` twice with the same ID, whereas this was possible with the previous API. This change basically improves the "encapsulation" of the IslandWindow class and prevents its (unsafe-use-) details from leaking.